src/llama-kv-cache.cpp

```diff
@@ -441,6 +441,13 @@ void llama_kv_cache_unified::defrag_sched(float thold) {
 
 void llama_kv_cache_unified::set_full() {
     n = size;
+
+    // when simulating a full KV cache, the specific value of the "head" pointer is not important because we are not
+    //   going to write any data - we just want to measure the memory needed by the graph in such state.
```

Comment:
Maybe I am misunderstanding, but setting it to zero should be necessary to get the biggest `KQ` possible, right? We would want every token in the cache to be used in the attention to estimate the worst case.

Reply:
The K*Q size is not determined by the `head`, but rather the `n` (or also referred to as `n_kv` in some places). The `head` is only used to offset the view into which we store the new K and V data from the current batch here:

https://github.com/ggml-org/llama.cpp/blob/9745d5fe4faa0ecfe883f6c4a1cf24e437c7a064/src/llama-graph.cpp#L1421-L1450

The `head` should never affect the shapes of the tensors - just the offsets of the K/V views.